addons/website_forum/static/src/js/website_forum.js

```diff
@@ -501,6 +510,57 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
         $('.forum_intro').slideUp();
         return true;
     },
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFlagValidatorClick: async function (ev) {
+        ev.preventDefault();
+        const currentTarget = ev.currentTarget;
+        const result = await this._rpc({
+            model: 'forum.post',
+            method: currentTarget.dataset.action,
+            args: [parseInt(currentTarget.dataset.postId)],
+        });
+        if (result) {
+            currentTarget.parentElement.classList.toggle('d-none');
+            const flaggedButton = currentTarget.parentElement.previousElementSibling,
+                child = flaggedButton.firstElementChild,
+                countFlaggedPosts = this.el.querySelector('#count_flagged_posts'),
+                count = parseInt(countFlaggedPosts.innerText, 10) - 1;
+
+            flaggedButton.innerText = _t(' Flag');
+            flaggedButton.prepend(child);
+            if (count === 0) {
+                countFlaggedPosts.classList.add("badge-light");
+            }
+            countFlaggedPosts.innerText = count;
+        }
+    },
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFlagMarkAsOffensiveClick: async function (ev) {
+        ev.preventDefault();
+        const result = await this._rpc({
+            route: $(ev.currentTarget).data('action'),
+        });
+        if (result) {
```

Comment:
How could result be falsy? Either it crashes and it handled at lower level (which shows a toaster) or it will just return something

+ as I asked, name if `template` not `result`